src/js/abstract/hocs/with-shadow.js

```diff
@@ -0,0 +1,18 @@
+const withShadow = Base =>
+  class WithShadow extends Base {
+    /**
+     * Attach shadow DOM upon connect.
+     *
+     * @param {String} [mode='open']
+     */
+    connectedCallback(mode = 'open') {
+      if (super.connectedCallback) {
+        super.connectedCallback();
+      }
+
+      const shadowRoot = this.attachShadow({ mode });
+      this._appendStyles(shadowRoot);
```

Comment:
@LucaMele 
How do we deal with appending styles for shadow DOM.
May styles should be an utility 🤔

Reply:
this is a bit of legacy code. We need to call the append style same way as `BaseComponent` and NOT like `BaseComponentGlobal`. Then styles will be automatically scoped inside the shadow dom

Reply:
aha, let's fix it then

Reply:
is `withStyles` obsolete then?

Reply:
i think so. It should do what base component does before becoming base component global

Reply:
this lines are important

const shadowRoot = this.attachShadow({ mode });
this._appendStyles(shadowRoot);

Reply:
i dont know now where the logic is that was in that function

Reply:
okay, let's check it together.
We will make it work like magic 💪